WordPress/src/main/java/org/wordpress/android/ui/stories/StoryMediaSaveUploadBridge.kt

```
@@ -0,0 +1,170 @@
+    // region Fields
+    private var job: Job = Job()
+
+    override val coroutineContext: CoroutineContext
+        get() = mainDispatcher + job
+
+    @Inject lateinit var editPostRepository: EditPostRepository
+
+    @Suppress("unused")
+    @OnLifecycleEvent(ON_CREATE)
+    fun onCreate(source: LifecycleOwner) {
+        EventBus.getDefault().register(this)
+    }
+
+    @Suppress("unused")
+    @OnLifecycleEvent(ON_DESTROY)
+    fun onDestroy(source: LifecycleOwner) {
+        // note: not sure whether this is ever going to get called if we attach it to the lifecycle of the Application
+        // class, but leaving it here prepared for the case when this class is attached to some other LifeCycleOwner
+        // other than the Application.
+        cancelAddMediaToEditorActions()
+        EventBus.getDefault().unregister(this)
+    }
+
+    // region Adding new composed / processed frames to a Story post
+    private fun addNewStoryFrameMediaItemsToPostAndUploadAsync(site: SiteModel, saveResult: StorySaveResult) {
+        // let's invoke the UploadService and enqueue all the files that were saved by the FrameSaveService
+        val frames = StoryRepository.getStoryAtIndex(saveResult.storyIndex).frames
+        val uriList = frames.map { Uri.fromFile(it.composedFrameFile) }
+        addNewMediaItemsToPostAsync(site, uriList)
+    }
+
+    private fun addNewMediaItemsToPostAsync(site: SiteModel, uriList: List<Uri>) {
+        // this is similar to addNewMediaItemsToEditorAsync in EditorMedia
+        launch {
+            addLocalMediaToPostUseCase.addNewMediaToEditorAsync(
+                    uriList,
+                    site,
+                    freshlyTaken = false, // we don't care about this
+                    editorMediaListener = this@StoryMediaSaveUploadBridge,
+                    doUploadAfterAdding = true
+            )
+            postUtils.preparePostForPublish(editPostRepository.getEditablePost()!!, site)
```

We try to avoid using the `!!` operator. Can we change this logic a bit to handle this null possibility? Especially if we don't expect the `post` to be null here. Wdyt?

---

I think if the Post is `null` at this point we'd rather crash than fail silently. There's not much we can do without a backing Post, so in this case the !! operator has intended meaning. As you said, given especially we don't expect things to be `null` here. Do you have something different in mind? I'm interested in understanding what other approaches exist 🙏 let me know @jd-alexander

---

Thanks for this @mzorz I am learning as I go along about how to handle these null values gracefully. What I have seen in other parts of WP Android is the usage of the `require*` method. Example: 

```kotlin
reportTransactionState(false)
            if (isUpdated) {
                requireNotNull(post).let {
                    _postChanged.value = Event(it)
         } }
```

Wdyt? 😄

---

I totally forgot about that one - even when I recently made similar changes in lots of places in 11934 🤕 - addressed in 2d813494bb

---

Awesome. Thanks 😄